client.go

```diff
@@ -462,13 +482,74 @@ func (c *Client) GetChannelUsers(name string) (*Users, error) {
 }
 
 /*
-AuthenticatePrivateChannel allows you to authenticate a users subscription to a
-private channel. It returns authentication signature to send back to the client
-and authorize them.
+AuthenticateUser allows you to authenticate a user s subscription to a
+private channel. It returns an authentication signature to send back to the client
+and authenticate them. In order to identify a user, this method acceps a map containing
+arbitrary user data. It must contain at least and id field with the user's id as a string.
 
 For more information see our docs: http://pusher.com/docs/authenticating_users.
 
-This is an example of authenticating a private-channel, using the built-in
+This is an example of authenticating a user, using the built-in
+Golang HTTP library to start a server.
+
+In order to authenticate a client, one must read the response into type `[]byte`
+and pass it in. This will return a signature in the form of a `[]byte` for you
+to send back to the client.
+
+	func pusherUserAuth(res http.ResponseWriter, req *http.Request) {
+
+		params, _ := ioutil.ReadAll(req.Body)
+		response, err := client.AuthenticateUser(params)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Fprintf(res, string(response))
+	}
+
+	func main() {
+		http.HandleFunc("/pusher/user-auth", pusherUserAuth)
+		http.ListenAndServe(":5000", nil)
+	}
+*/
+func (c *Client) AuthenticateUser(params []byte, userData map[string]interface{}) (response []byte, err error) {
+	socketID, err := parseUserAuthenticationRequestParams(params)
+	if err != nil {
+		return
+	}
+
+	if err = validateSocketID(&socketID); err != nil {
+		return
+	}
+
+	if err = validateUserData(userData); err != nil {
+		return
+	}
+
+	var _jsonUserData []byte
```

Comment:
move 529-535 to the separate function, it will make authentication logic a bit more clear in this function.